addons/stock_calendar/models/procurement.py

```diff
@@ -52,13 +47,13 @@ def _get_purchase_order_date(self, cr, uid, procurement, company, schedule_date,
         """
         if procurement.next_purchase_date:
             return datetime.strptime(procurement.next_purchase_date, DEFAULT_SERVER_DATETIME_FORMAT)
-        return super(procurement_order, self)._get_purchase_order_date(cr, uid, procurement, company, schedule_date, context=context)
+        return super(ProcurementOrder, self)._get_purchase_order_date(cr, uid, procurement, company, schedule_date, context=context)
 
     @api.v8
     def _get_purchase_schedule_date(self):
         if self.next_delivery_date:
-            return datetime.strptime(self.next_delivery_date, DEFAULT_SERVER_DATETIME_FORMAT)
-        return super(procurement_order, self)._get_purchase_schedule_date()
+            return fields.Datetime.from_string(self.next_delivery_date)
+        return super(ProcurementOrder, self)._get_purchase_schedule_date()
 
     @api.v7
     def _get_purchase_schedule_date(self, cr, uid, procurement, context=None):
```

Comment:
Why you need @api.v7 and @api.v8 both methods ?. You can decorate a method with @api.cr_uid_records_context decorator .